qiskit/providers/aer/backends/cluster/clusterjobset.py

```diff
@@ -0,0 +1,236 @@
+    :meth:`AerClusterManager.run()`.
+    It provides methods that allow you to interact
+    with the jobs as a single entity. For example, you can retrieve the results
+    for all of the jobs using :meth:`results()` and cancel all jobs using
+    :meth:`cancel()`.
+    """
+
+    def __init__(self, experiments: List[QasmQobj], name: Optional[str] = None):
+        """JobSet constructor.
+
+        Args:
+            experiments: List[QasmQobjs] to execute.
+            name: Name for this set of jobs.
+        """
+        self._name = name or str(uuid.uuid4())
+        self._experiments = experiments
+
+        # Used for caching
+        self._future = None
+        self._futures = []
+        self._results = None
+        self._error_msg = None
+
+    def run(self,
+            executor: futures.Executor,
+            backend: AerBackend,
+            *run_args: Any,
+            **run_kwargs: Any) -> None:
+        """Execute this set of jobs on an executor.
+
+        Args:
+            executor: The executor used to submit jobs asynchronously.
+            backend: The backend to run the jobs on.
+            *run_args: Positional arguments passed through to backend.run.
+            **run_kwargs: Keyword arguments passed through to backend.run.
+
+        Raises:
+            RuntimeError: If the jobs were already submitted.
+        """
+        if self._futures:
+            raise RuntimeError(
+                'The jobs for this managed job set have already been submitted.')
+
+        self._future = True
+        total_jobs = len(self._experiments)
+        for i, exp in enumerate(self._experiments):
+            cjob = CJob(backend, exp, *run_args, **run_kwargs)
+            cjob.submit(executor=executor)
+            logger.debug("Job %s submitted", i + 1)
+            self._futures.append(cjob)
+
+    @requires_submit
+    def statuses(self) -> List[Union[JobStatus]]:
+        """Return the status of each job in this set.
+
+        Returns:
+            A list of job statuses.
+        """
+        return [cjob.status() for cjob in self._futures]
+
+    @requires_submit
+    def result(
+            self,
+            timeout: Optional[float] = None,
+            raises: Optional[bool] = False
+    ) -> Result:
+        """Return the results of the jobs as a single Result object.
+
+        This call will block until all job results become available or
+        the timeout is reached.
+
+        Args:
+           timeout: Number of seconds to wait for job results.
+           raises: whether or not to re-raise exceptions thrown by jobs
+
+        Returns:
+            qiskit.Result: Result object
+
+        Raises:
+            AerClusterTimeoutError: if unable to retrieve all job results before the
+                specified timeout.
+
+        """
+        return self.results(timeout, raises).combine_results()
+
+    @requires_submit
+    def results(
```

Comment:
This could be `worker_result` and take a worker index as its argument.
ie `job.worker_result(i)` and each worked should return a regular qiskit `Result` object from this function.